src/updates.ts

```diff
@@ -185,17 +184,65 @@ async function showUpdateReadyToInstallNotice() {
 	}
 }
 
-async function showInstallLocationErrorNotice() {
+async function showInstallLocationErrorNotice( message: string ) {
 	if ( process.platform === 'darwin' ) {
 		await dialog.showMessageBox( {
 			type: 'warning',
 			buttons: [ __( 'OK' ) ],
 			title: __( 'Could Not Update Studio' ),
 			message:
+				message ||
 				// Translators: "Applications" is the name of the folder apps are installed to on macOS
 				__(
 					'Studio can only update automatically from the Applications folder. Please move Studio to Applications and try again.'
 				),
 		} );
 	}
 }
+
+function isAppRunningFromDMG(): boolean {
+	if ( process.platform !== 'darwin' ) {
+		return false;
+	}
+
+	const appPath = app.getPath( 'exe' );
+	return appPath.startsWith( '/Volumes/' );
+}
+
+function handleReadOnlyVolumeError( err: Error ) {
```

Comment:
Could we reduce responsibility by moving Sentry logging out of this function and renaming it to `showReadOnlyVolumeError`?

Reply:
Also, I think we can merge it with `showInstallLocationErrorNotice` as it seems like an unnecessary abstraction now.

Reply:
@wojtekn, added all changes proposed by you, makes sense 👍